src/util/output_channels.ts

```diff
@@ -45,6 +55,81 @@ export function openNamedOutputChannel(name?: string): vscode.OutputChannel | un
   return channel;
 }
 
+async function credHelper(startProcess: ChildProcess): Promise<void> {
+  const cred = await getHelpers();
+  const resourceGroups: vscode.QuickPickItem[] = cred.map((label) => ({ label }));
+  const credentialHelper = await multiStep.showQuickPick({
+    title: 'Select credentials helper',
+    placeholder: 'Choose credentials helper',
+    items: resourceGroups,
+  });
+  if (!credentialHelper) {
+    startProcess.stdin.end();
+    return null;
+  }
+  if (credentialHelper.label === 'None') {
+    startProcess.stdin.write(`\n`);
+  } else {
+    startProcess.stdin.write(`${credentialHelper.label}\n`);
+  }
+}
+
+async function getUsernameOrPassword(
+  message: string,
+  prompt: string,
+  placeholder: string,
+  passwordType?: boolean,
+  errorMessage?: string,
+  reattemptForLogin?: boolean,
+): Promise<string | null> {
+  return multiStep.showInputBox({
+    title: message,
+    prompt,
+    placeholder,
+    reattemptForLogin,
+    validate: (value: string) => {
+      if (validator.isEmpty(value)) {
+        return errorMessage;
+      }
+      return null;
+    },
+    password: passwordType,
+  });
+}
+
+async function provideUserNameAndPassword(
+  startProcess: ChildProcess,
+  message: string,
+  reattemptForLogin?: boolean,
+): Promise<void> {
+  const userInfo = 'Provide username for image registry.';
+  const userName = await getUsernameOrPassword(
+    message,
+    userInfo,
+    userInfo,
```

Comment:
I would add a different placeholder instead of copying the same message. `Enter the username/password to use`